tests/admin_changelist/tests.py

```diff
@@ -1067,6 +1068,17 @@ def test_link_field_display_links(self):
             "documents/blues_history.txt</a>" % g.pk,
         )
 
+    def test_blank_str_display_links(self):
+        self.client.force_login(self.superuser)
+        p = Photographer.objects.create(name="Antoliny")
+        response = self.client.get(
+            reverse("admin:admin_changelist_photographer_changelist")
+        )
+        self.assertContains(
+            response,
+            '<a href="/admin/admin_changelist/photographer/%s/change/">-</a>' % p.pk,
```

Comment:
We should try to reuse models where we can 
```suggestion
        gc = GrandChild.objects.create(name="           ")
        response = self.client.get(
            reverse("admin:admin_changelist_grandchild_changelist")
        )
        self.assertContains(
            response,
            '<a href="/admin/admin_changelist/grandchild/%s/change/">-</a>' % gc.pk,
```

This however fails as this patch only fixes the __str__ case. We probably want something like:
```diff
--- a/django/contrib/admin/templatetags/admin_list.py
+++ b/django/contrib/admin/templatetags/admin_list.py
@@ -223,11 +223,11 @@ def items_for_result(cl, result, form):
             empty_value_display = getattr(
                 attr, "empty_value_display", empty_value_display
             )
+            if isinstance(value, str) and value.strip() == "":
+                value = ""
             if f is None or f.auto_created:
                 if field_name == "action_checkbox":
                     row_classes = ["action-checkbox"]
-                elif field_name == "__str__":
-                    value = value.strip()
                 boolean = getattr(attr, "boolean", False)
```